lib/krane/resource_deployer.rb

```diff
@@ -128,17 +132,15 @@ def deploy_resources(resources, prune: false, verify:, record_summary: true)
       end
     end
 
-    def apply_all(resources, prune)
+    def apply_all(resources, prune, dry_run: false)
```

Comment:
They are simply included in the `apply_all` call. From what I observed, it works as expected, and the actual `dry-run=false` invocation will use the appropriate deploy method. This seemed like the best way to capture all resources in batch validation. One thing this won't catch is resources that exclusively use `:create` (without, e.g. a fallback to `:replace`), since that will pass validation but fail on deploy if the resource already exists in cluster. IMO that's a bug in their config, not Krane, though.

Reply:
How hard is it to special case this e.g. do the batch and then dry run the create only resources too? This seems like a nice to catch (though we'd only catch that today for secrets).

Reply:
Opened #787 to keep track of this refinement.